### DISCUSSION TITLE
How to get a system's current timezone

### DISCUSSION THREAD

**Discussion by amboyconnector** [CrowdStrike/falconpy#685] [Q&A] [answered]
I'm going to use the API to download some records from our systems, whose timestamps will always be in those systems' local timezone, and they could be anywhere in the world. I'll want to adjust each timestamp to GMT before processing them with other code. Is there a way to determine the timezone a host is in at the time of an API query, so that I can make that adjustment? 

Alternately, If there isn't a timezone property available via the API, is there a timestamp available somewhere that will be the system's current local time, so I could calculate the timezone manually?

**Comment by jshcodes** [comment, marked as answer]
This detail is not returned as part of a `GetDeviceDetails` call, but you could grab it from RTR if you're wanting to use the API.

Our sample located [here](https://github.com/CrowdStrike/falconpy/blob/main/samples/rtr/bulk_execute.py) will execute a command across a batch of hosts.  (This example does not demonstrate queueing.)

Something along the lines of:
```shell
python3 bulk_execute.py -k API_KEY_HERE -s API_SECRET_HERE -f HOSTNAME_SEARCH_STRING -c date
```

**Reply by amboyconnector** [reply]
I'll be connecting via RTR anyway, so that's an easy solution - thanks!

**Reply by jshcodes** [reply]
For just the timezone you can try these:

# Windows

```shell
python3 bulk_execute.py -k API_KEY_HERE -s API_SECRET_HERE -f HOSTNAME_SEARCH_STRING -c "Get-TimeZone"
```

# Linux

```shell
python3 bulk_execute.py -k API_KEY_HERE -s API_SECRET_HERE -f HOSTNAME_SEARCH_STRING -c "date '%Z %z'"
```

😄
